pkg/container-utils/containerd/containerd.go

```diff
@@ -15,31 +15,293 @@
 package containerd
 
 import (
+	"context"
+	"fmt"
 	"time"
 
-	criclient "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/cri"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
 	runtimeclient "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/runtime-client"
 )
 
 const (
 	DefaultTimeout = 2 * time.Second
+
+	LabelK8sContainerName         = "io.kubernetes.container.name"
+	LabelK8sContainerdKind        = "io.cri-containerd.kind"
+	LabelK8sContainerdKindSandbox = "sandbox"
 )
 
+var log = logrus.WithField("container-client", "containerd")
+
 type ContainerdClient struct {
-	criclient.CRIClient
+	client *containerd.Client
+	ctx    context.Context
 }
 
 func NewContainerdClient(socketPath string) (runtimeclient.ContainerRuntimeClient, error) {
 	if socketPath == "" {
 		socketPath = runtimeclient.ContainerdDefaultSocketPath
 	}
 
-	criClient, err := criclient.NewCRIClient(runtimeclient.ContainerdName, socketPath, DefaultTimeout)
+	dialCtx, cancelFunc := context.WithTimeout(context.TODO(), DefaultTimeout)
+	defer cancelFunc()
+	grpcConn, err := grpc.DialContext(
+		dialCtx,
+		"unix://"+socketPath,
```

Comment:
```suggestion
		"unix:"+socketPath,
```
How about using ["unix:"](https://github.com/grpc/grpc/blob/master/doc/naming.md?plain=1#L37) so it works for both absolute and relative paths like before?